src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -1772,16 +1772,18 @@ private void OnPortPropertyChanged(object sender, PropertyChangedEventArgs args)
 
         private void OnPortConnected(PortModel port, ConnectorModel connector)
         {
-            var handler = PortConnected;
-            if (null != handler) handler(port, connector);
-
+          
             if (port.PortType != PortType.Input) return;
 
             var data = InPorts.IndexOf(port);
             var startPort = connector.Start;
             var outData = startPort.Owner.OutPorts.IndexOf(startPort);
             ConnectInput(data, outData, startPort.Owner);
             startPort.Owner.ConnectOutput(outData, data, this);
+
+            var handler = PortConnected;
```

Comment:
this event must be raised after the InputData is set in the ConnectInput call above.

Reply:
more info, if this is not done in this order it's possible that a watch node will get the previous value from a node.

Reply:
@mjkkirschner Deja-vu...